library/packages/src/modules/PackageSystem.rb

```diff
@@ -107,11 +105,8 @@ def EnsureSourceInit
         EnsureTargetInit()
       end
 
-      sources = Pkg.SourceStartCache(true)
-
-      if Ops.greater_than(Builtins.size(sources), 0)
-        @source_initialized = true
-      else
+      # at least one enabled repository?
+      if Pkg.SourceGetCurrent(true).empty?
```

Comment:
... or is the magic done by this true / false parameter in `SourceGetCurrent`? May be it would deserve better name then

Reply:
That boolean parameter just defines whether the result should include all repositories or just enabled ones.

Actually the `Pkg.SourceStartCache` call is the same as `Pkg.SourceStartManager + Pkg.SourceGetCurrent` (see [here](https://github.com/yast/yast-pkg-bindings/blob/987c4933a9fa8935664bd35d08d4c5ce5d050e30/src/Source_Load.cc#L545-L550)).

Yes, the naming and that boolean parameter look strange, but that's a legacy from even the pre-libzypp era...

Reply:
hmm ... i was thinking of some constants ... but that might be an overkill.

Reply:
A constant with some self-explanatory name or at least a comment above the line would definitely help in the future. We'll be having issues when refactoring the code. $0.02